### PR TITLE
Changed set_mode to work with kx3

### PR DESCRIPTION
**Pull request by n9seo**
Hi,

This worked for me so I can use rigctl to set_mode as the following:
PKTLSB = Sets the rig to MODE DATA REV (MD9) and SUBMODE DATA A (DT0)
RTTY = will set to the former PKTLSB RTTY DT1 SUBMODE AFSK A for RTTY

So we can send PKTLSB and set the rig to HF Packet mode for soundcard/TNCs.  
RTTY mode will set the KX3 to RTTY Mode.

I am not sure how this will impact K3's but it seems to work for KX3.  

Would be nice to add other case statements for the other submodes but I am not sure how that would interact with hamlib/rigctl presently.

### CODE REVIEW

**Review by mdblack98 — COMMENTED** [review]
PKTLSB should be RTTY and PKTUSB should be RTTYR
Should be testable if you can look at your own transmission but that would take a 2nd rig.
PKTLSB should produce a tuning tone on lower side band and RTTY should also produce a signal on lower side band.

And I'm not convinced RTTY modes fall under DT2.

**Comment by mdblack98** [comment]
RTTY is LSB and RTTYR is USB....need to change those.

Mike W9MDB

**Comment by n9seo** [comment]
Hi,

Looking at it from the Rig end DT0 is what you'd want for typical soundmodems/fldigi soundcard type streams.   DT1/2/3 I am not sure what to do with yet.   You got I think afsk rtty and fsk rtty and a psk mode that all technically can be our normal modes and have a 'reverse' mode technically speaking.  PSK  has a 31 and 63 bps option but I havent looked at that yet.

I need to review this again for the RTTY/PKTUSB thing.   I tried it last RTTY = USB and RTTYR was LSB on DT0.  So thats what I've been running since.   It seems backwards to me but it worked.  So let me just recheck it again.

**Comment by mdblack98** [comment]
RTTY on LSB and RTTYR on USB is an historical thing.
Original RTTY was always LSB -- then USB was introduced and RTTYR became the new name for it.  It just stands for REVERSED and that term has no implied correlation to upper/lower sideband.
Mike W9MDB

 

    On Tuesday, March 2, 2021, 04:50:10 PM CST, n9seo <notifications@github.com> wrote:  
 
 


Hi,

Looking at it from the Rig end DT0 is what you'd want for typical soundmodems/fldigi soundcard type streams. DT1/2/3 I am not sure what to do with yet. You got I think afsk rtty and fsk rtty and a psk mode that all technically can be our normal modes and have a 'reverse' mode technically speaking. PSK has a 31 and 63 bps option but I havent looked at that yet.

I need to review this again for the RTTY/PKTUSB thing. I tried it last RTTY = USB and RTTYR was LSB on DT0. So thats what I've been running since. It seems backwards to me but it worked. So let me just recheck it again.

**Comment by n9seo** [comment]
Hi,

I could not get `RIG_MODE_RTTY` here to set the right to LSB until I changed that to RTTYR.   Could not determine why that was from here.

```        case RIG_MODE_PKTLSB:
        mode = RIG_MODE_RTTY;
        snprintf(cmd_m, sizeof(cmd_m),
                 "DT0"); /* DATA A mode - DATA (REV) on LSB optimized for HF Packet, VFO dial is suppressed carrier QRG */
        break;
```

**Comment by n9seo** [comment]
Sorry, I could not get it to put the RIG into DATA SUBMODE A on LSB until I used RTTYR

**Comment by mdblack98** [comment]
From the Elecraft manual for example...
K21, K2 Normal/rtty_off: Same as K20, except that MD and IF report RTTY and RTTY-reverse modes as LSBand USB, respectively. This may be useful with programs that don’t support a separate RTTY mode.
So...
RTTY=DATARRTTYR=DATA
Mike W9MDB
 

    On Tuesday, March 2, 2021, 04:59:07 PM CST, n9seo <notifications@github.com> wrote:  
 
 


Sorry, I could not get it to put the RIG into DATA SUBMODE A on LSB until I used RTTYR

File: rigs/kenwood/k3.c [hunk 1]
```diff
@@ -1066,28 +1066,39 @@ int k3_set_mode(RIG *rig, vfo_t vfo, rmode_t mode, pbwidth_t width)
 
     switch (mode)
     {
-    case RIG_MODE_PKTLSB:
-        mode = RIG_MODE_RTTY;
+        case RIG_MODE_PKTLSB:
+        mode = RIG_MODE_RTTYR;
```
Comment (n9seo) [inline rigs/kenwood/k3.c:1070]: For some reason this had to be RTTYR for rigctl/set_mode PKTLSB 0 to send the rig to DATA REV submode data A (DT0)

**Comment by mdblack98** [comment]
I see your problem now...Turns out the modes were backwards in kenwood.c mode_tableI just fixed that.
Mike W9MDB
 

    On Tuesday, March 2, 2021, 06:48:49 PM CST, n9seo <notifications@github.com> wrote:  
 
 


@n9seo commented on this pull request.

In rigs/kenwood/k3.c:
> +        case RIG_MODE_PKTLSB:
+        mode = RIG_MODE_RTTYR;

For some reason this had to be RTTYR for rigctl/set_mode PKTLSB 0 to send the rig to DATA REV submode data A (DT0)

**Comment by n9seo** [comment]
Hi,
Thanks.  I think that would explain my opposite thing. I was convinced it was doing that somewhere.  I need to go look at that now.  Prior to that last night before I got too tired to carry on I noticed from the radios panel that DT0 and DT3 share that MD6 = USB and the other two DT modes start as LSB on MD6. If you wanted the opposite you send MD9 and DT0.  This is if you were setting it from the  panel controls that each submode (DT cmd) determines the default MD6's sideband mode.     

-k

**Comment by mdblack98** [comment]
The CAT manual says that too.  Thought I'm not quite understanding what they mean about alternating between MD6 and MD9.
This RTTY stuff doesn't get exercised a lot so you're the first person to notice a problem.
The norm/rev conditions for the K3’s data sub-modes arehandled in two pairs at present: DATA A/PSK D, and AFSK A/FSK D. E.g., if the radio is set up for DATA Amode, alternating between MD6 and MD9 will cause both DATA A and PSK D to be set to the samenormal/reverse condition.
Mike W9MDB

 

    On Wednesday, March 3, 2021, 06:38:21 AM CST, n9seo <notifications@github.com> wrote:  
 
 


Hi,
Thanks. I think that would explain my opposite thing. I was convinced it was doing that somewhere. I need to go look at that now. Prior to that last night before I got too tired to carry on I noticed from the radios panel that DT0 and DT3 share that MD6 = USB and the other two DT modes start as LSB on MD6. If you wanted the opposite you send MD9 and DT0. This is if you were setting it from the panel controls that each submode (DT cmd) determines the default MD6's sideband mode.

-k

**Comment by n9seo** [comment]
Hi,

I *think* I got this in order.  Please check when you can.

**Comment by mdblack98** [comment]
This has a conflict in the merge....
Can you check please?
Mike W9MDB

 

    On Wednesday, March 3, 2021, 08:01:29 AM CST, n9seo <notifications@github.com> wrote:  
 
 


Hi,

I think I got this in order. Please check when you can.

**Comment by n9seo** [comment]
OOps,

Let me try to resolve this.  I may redo this PR this evening after work.

**Comment by n9seo** [comment]
Hi,

I think I resolved the conflicts, at least according to Github.

**Comment by mdblack98** [comment]
You've got 11 commits queued up which most are broken.
Pleases re-do your K3 patch on a new fork or such.

**Comment by n9seo** [comment]
Alright, sorry about that.  WIll do. I tried to do the simple delete fork and refork this AM but had to open a github ticket since GH is not behaving the same as it does in Enterprise mode.  They may have to perm delete the old one.  As soon as thats done and I can I will repost the modified k3.c.

**Pull request closed without merging**